### PR TITLE
Squash db schema version 10

### LINKED ISSUE #2200: squash sql schema changes up to schema version 10
Database schema version 10 was introduced in version 11.0.0 and now that all nodes are running versions more recent than this, we should "squash" all schema upgrades.

This would allow us to remove some of the confusing code in Database.cpp like https://github.com/stellar/stellar-core/blob/master/src/database/Database.cpp#L212

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
# Description

Resolves #2200 

This PR consolidates database changes to the version of the schema introduced in version 11.0.0
(as the network has been running protocol 11 for a while, it is safe to assume that upgrades from versions 10.x or older don't need to be supported anymore)

### CODE REVIEW

File: src/ledger/LedgerTxnAccountSQL.cpp [hunk 1]
```diff
@@ -530,15 +530,18 @@ LedgerTxnRoot::Impl::dropAccounts()
...
-           "homedomain      VARCHAR(32)  NOT NULL,"
-           "thresholds      TEXT         NOT NULL,"
-           "flags           INT          NOT NULL,"
-           "lastmodified    INT          NOT NULL"
+           "accountid          VARCHAR(56)  PRIMARY KEY,"
+           "balance            BIGINT       NOT NULL CHECK (balance >= 0),"
+           "buyingliabilities  BIGINT CHECK (buyingliabilities >= 0),"
+           "sellingliabilities BIGINT CHECK (sellingliabilities >= 0),"
+           "seqnum             BIGINT       NOT NULL,"
+           "numsubentries      INT          NOT NULL CHECK (numsubentries >= "
```
Comment (marta-lokhova) [inline src/ledger/LedgerTxnAccountSQL.cpp]: why the formatting change?

File: src/ledger/LedgerTxnAccountSQL.cpp [hunk 2]
```diff
@@ -541,6 +473,7 @@ LedgerTxnRoot::Impl::dropAccounts()
            "homedomain         VARCHAR(32)  NOT NULL,"
            "thresholds         TEXT         NOT NULL,"
            "flags              INT          NOT NULL,"
+           "signers            TEXT,"
            "lastmodified       INT          NOT NULL"
            ");";
     mDatabase.getSession() << "CREATE TABLE signers"
```
Comment (marta-lokhova) [inline src/ledger/LedgerTxnAccountSQL.cpp]: I don't think we need this or the index creation code below.

File: src/database/Database.cpp [hunk 3]
```diff
@@ -189,15 +189,11 @@ Database::applySchemaUpgrade(unsigned long vers)
...
-        // add tracking table information
-        mApp.getHerderPersistence().createQuorumTrackingTable(mSession);
-        break;
     default:
-        if (vers <= 9)
+        if (vers <= 10)
         {
             throw std::runtime_error(
-                "Database version is too old, must use at least 10");
+                "Database version is too old, must use at least 11");
```
Comment (marta-lokhova) [inline src/database/Database.cpp]: but version 11 doesn't exist yet :) `applySchemaUpgrade` should just be a no-op if `vers == SCHEMA_VERSION`

File: src/ledger/LedgerTxnAccountSQL.cpp (hunk 1 above)
Comment (MonsieurNicolas) [reply src/ledger/LedgerTxnAccountSQL.cpp]: column shift

File: src/ledger/LedgerTxnAccountSQL.cpp (hunk 2 above)
Comment (MonsieurNicolas) [reply src/ledger/LedgerTxnAccountSQL.cpp]: good catch

File: src/database/Database.cpp (hunk 3 above)
Comment (MonsieurNicolas) [reply src/database/Database.cpp]: Easier to maintain like this; that said, let me take out the hardcoded 11

**Comment by MonsieurNicolas** [comment]
updated @marta-lokhova

**Comment by marta-lokhova** [comment]
r+ b36409a5bf0e6d40d0a5b458adebef29c2b05613

**Pull request merged**